Add "Start dragging object" to enable custom picking of object to be dragged

tristanbob opened the pull request:
Default dragging requires the cursor to be over an object. This is not great for fabrics of small objects, because there is a lot of space that will not trigger a drag to start.  This change adds a new action to manually start dragging, so game creators can choose when and which objects to grab.

To enable this logic, I added a new option to the Mouse Button property: "None (disable automatic dragging)".
Do you like this implementation, or should I make a separate property, such as: "Enable automatic dragging"?

![image](https://user-images.githubusercontent.com/8879811/236732667-7de8c521-ee20-44d9-ac00-51b9ee4ad065.png)

## Example game

https://gd.games/victrisgames/building-monsters

tristanbob commented:
Note: I'm repeating this question so it is more obvious that I am seeking feedback.

I added a new option to the `Mouse Button` property: `None (disable automatic dragging)`
Do you like this implementation, or should I make a separate property, such as: `Enable/Disable automatic dragging`?

tristanbob commented:
Any feedback on my question?  Which is the best implementation:

- Add a new option to the `Mouse Button` property: `None (disable automatic dragging)`
OR
- Add new property `Enable/Disable automatic dragging`

D8H commented:
> Any feedback on my question? Which is the best implementation:
> 
>     * Add a new option to the `Mouse Button` property: `None (disable automatic dragging)`
>       OR
> 
>     * Add new property `Enable/Disable automatic dragging`

I'm not sure the 1st option is possible because the "mouse button" parameter type can't have this new choice in the action to change the property.

tristanbob commented:
> > Any feedback on my question? Which is the best implementation:
> 
> > 
> 
> >     * Add a new option to the `Mouse Button` property: `None (disable automatic dragging)`
> 
> >       OR
> 
> > 
> 
> >     * Add new property `Enable/Disable automatic dragging`
> 
> 
> 
> I'm not sure the 1st option is possible because the "mouse button" parameter type can't have this new choice in the action to change the property.

This is actually what my commit does. I'm using a "list of strings".

D8H commented:
> This is actually what my commit does. I'm using a "list of strings".

Yes, but, ideally the "mouse button" parameter type should be used because the editor can make a better UI with translations and icons for instance.
The same types could be added to properties in the future as it seems useful for input mapping behaviors.

tristanbob commented:
> > This is actually what my commit does. I'm using a "list of strings".
> 
> Yes, but, ideally the "mouse button" parameter type should be used because the editor can make a better UI with translations and icons for instance. The same types could be added to properties in the future as it seems useful for input mapping behaviors.

OK, I think what you are saying is:

1) There are no current "mouse button" parameter types, but in the future, there might be.
2) Changing the list of strings to match the standard "mouse button" format will make it easier to transition to the above.
3) Therefore, I should create a new property to enable/disable automatic dragging 

Did I understand correctly?

![image](https://github.com/GDevelopApp/GDevelop-extensions/assets/8879811/b92d0820-cd9f-4678-97db-4f7679edd7f7)

tristanbob commented:
I've changed this to a new property and removed the option on the mouse buttons.

Next, I will create an update to the existing example game for "draggable physics" to test this change.

tristanbob commented:
I tested this on the default example, but I couldn't think of a good way to add it to the game that made sense.
